File: briar-android/src/main/java/org/briarproject/briar/android/telegram/TelegramConversationMapper.java
package org.briarproject.briar.android.telegram;

import org.briarproject.briar.api.telegram.TelegramMessage;
import org.briarproject.nullsafety.NotNullByDefault;

import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;

@NotNullByDefault
final class TelegramConversationMapper {

	private static final Comparator<TelegramConversationUiMessage>
			BY_DATE_ASCENDING =
			Comparator.comparingLong(TelegramConversationUiMessage::getDateMillis)
					.thenComparingLong(
							TelegramConversationUiMessage::getMessageId);

	static List<TelegramConversationUiMessage> toUiMessages(
			List<TelegramMessage> messages) {
		List<TelegramConversationUiMessage> items =
				new ArrayList<>(messages.size());
		for (TelegramMessage message : messages) {
			if (message.getText().isEmpty()) continue;
			items.add(new TelegramConversationUiMessage(
					message.getMessageId(),
					message.getDateSeconds() * 1000L,
					message.isOutgoing(),
					message.getText()
			));
		}
		items.sort(BY_DATE_ASCENDING);
		return items;
	}
}

File: briar-android/src/main/kotlin/org/briarproject/briar/android/telegram/TelegramConversationMapper.kt
package org.briarproject.briar.android.telegram

import org.briarproject.briar.android.connector.ConnectorConversationMessageItem
import org.briarproject.briar.api.connector.ConnectorMessage

object TelegramConversationMapper {

	private val byDateAscending = compareBy<ConnectorMessage> {
		it.dateSeconds
	}.thenBy {
		it.sourceMessageOrder
	}.thenBy {
		it.messageId
	}

	@JvmStatic
	fun toUiMessages(
		messages: List<ConnectorMessage>,
	): List<ConnectorConversationMessageItem> {
		val items = ArrayList<ConnectorConversationMessageItem>(messages.size)
		for (message in messages.sortedWith(byDateAscending)) {
			val text = message.text
			if (text.trim().isEmpty()) continue
			items.add(ConnectorConversationMessageItem.from(message))
		}
		return items
	}
	}
